nat.go

```diff
@@ -25,3 +31,211 @@ type NAT struct {
 	LogicalPort string
 	ExternalID  map[interface{}]interface{}
 }
+
+func (odbi *ovndb) rowToNat(uuid string) *NAT {
+	cacheNAT, ok := odbi.cache[tableNAT][uuid]
+	if !ok {
+		return nil
+	}
+
+	nat := &NAT{
+		UUID:       uuid,
+		Type:       cacheNAT.Fields["type"].(string),
+		ExternalIP: cacheNAT.Fields["external_ip"].(string),
+		LogicalIP:  cacheNAT.Fields["logical_ip"].(string),
+		ExternalID: cacheNAT.Fields["external_ids"].(libovsdb.OvsMap).GoMap,
+	}
+
+	if mac, ok := cacheNAT.Fields["external_mac"]; ok {
+		switch mac.(type) {
+		case libovsdb.UUID:
+			nat.ExternalMAC = mac.(libovsdb.UUID).GoUUID
+		case string:
+			nat.ExternalMAC = mac.(string)
+		default:
+		}
+	}
+
+	if lip, ok := cacheNAT.Fields["logical_port"]; ok {
+		switch lip.(type) {
+		case libovsdb.UUID:
+			nat.LogicalIP = lip.(libovsdb.UUID).GoUUID
+		case string:
+			nat.LogicalIP = lip.(string)
+		default:
+		}
+
+	}
+
+	return nat
+}
+
+func (odbi *ovndb) lrNatAddImp(lr string, Type string, externalIp string, externalMac string, logicalIp string, logicalPort string, external_ids map[string]string) (*OvnCommand, error) {
```

Comment:
@vtolstov this place Type can change to ntype too

Reply:
Thanks, you can do that too:)

Reply:
This time in china is weekend lol , I leave my  computer for code at company.

Reply:
done